kore/src/Kore/Internal/MultiOr.hs

```diff
@@ -194,7 +194,7 @@ distributeAnd =
     foldr (crossProductGeneric and') (singleton MultiAnd.top)
   where
     and' term ma =
-        term : MultiAnd.extractPatterns ma & MultiAnd.make
+        term : Foldable.toList ma & MultiAnd.make
```

Comment:
This would leave our options open in the future to pick a more efficient implementation:

```suggestion
        MultiAnd.singleton term <> ma
```

Reply:
Wow, so this change made one of the `distributeAnd` unit tests fail, and it's because we're deriving the `Semigroup` (and `Monoid`) instance for `MultiAnd`. It's good that we caught this now.